codeclimate-wrapper

```diff
@@ -18,32 +18,52 @@ EOF
   exit 1
 }
 
+docker_for_mac() {
+  test -S /var/tmp/docker.sock > /dev/null 2>&1
+}
+
+socket_exists() {
+  test -S $socket_path > /dev/null 2>&1
+}
+
+socket_path() {
```

Comment:
>If the current version is symlinking to `/var/run/docker.sock` we should just use that and not bother with an alternate `$socket_path`

The current script tests whether this file is a socket file, which fails as it's a symlink.

> If the documententation directs users to unset `$DOCKER_`, we should rely on that and not worry about short-circuiting before checking for docker-machine

Agreed. 👍 

> This only leaves the potential for a user having `boot2docker` on `$PATH`, but actually intending to use Docker Machine -- I think it's OK not to handle this edge-case since boot2docker is deprecated

That sounds like an argument for removing that test altogether, but I'm not really fussed either way.